lib/Conversion/TorchToLinalg/Pooling.cpp

```diff
@@ -583,6 +584,264 @@ class ConvertAtenMaxPool2dWithIndicesOp
+        Value boundIndex = rewriter.create<arith::SubIOp>(
+            loc, arg, getValueOrCreateConstantIndexOp(rewriter, loc, low[i]));
+        Value cmp1 = rewriter.create<arith::CmpIOp>(
+            loc, arith::CmpIPredicate::sgt, boundIndex, bound);
+        Value cmp2 = rewriter.create<arith::CmpIOp>(
+            loc, arith::CmpIPredicate::slt, boundIndex, zero);
+        boundIndex =
+            rewriter.create<arith::SelectOp>(loc, cmp1, bound, boundIndex);
+        boundIndex =
+            rewriter.create<arith::SelectOp>(loc, cmp2, zero, boundIndex);
+        indices.emplace_back(boundIndex);
+      }
+
+      Value borderVal = rewriter.create<tensor::ExtractOp>(loc, src, indices);
+      rewriter.create<tensor::YieldOp>(loc, borderVal);
+      return pad.getResult();
+    };
+
+    if (expectedInputShape != selfType.getShape()) {
+      // Input tensor sizes are smaller than output size, pad inputs with border
+      // values.
+      // TODO: this is probably expensive, and it may be possible to solve by
+      // cleverly constructing affine maps for the next linalg.generic op,
+      // but I'm not smart enough to figure this out.
+
+      SmallVector<OpFoldResult> low(outRank, rewriter.getI64IntegerAttr(0));
+      SmallVector<OpFoldResult> high;
+      for (auto &&[i, inpSize, outSize] :
+           llvm::enumerate(selfType.getShape(), expectedInputShape)) {
+        if (int64_t(i) < NC) {
+          high.emplace_back(rewriter.getI64IntegerAttr(0));
+          continue;
+        }
+        high.emplace_back(rewriter.getI64IntegerAttr(outSize - inpSize));
+      }
+
+      self = padBorder(self, low, high);
+      indices = padBorder(indices, low, high);
+    }
+
+    Value init = rewriter.create<tensor::EmptyOp>(
+        loc, getAsOpFoldResult(outSizePadded), selfType.getElementType());
+
+    SmallVector<AffineExpr> inputExprs;
+    SmallVector<AffineExpr> outputExprs;
+    for (auto i : llvm::seq<int64_t>(0, outRank)) {
+      AffineExpr dim = rewriter.getAffineDimExpr(i);
+      if (i < NC) {
+        inputExprs.emplace_back(dim);
+      } else {
+        int64_t j = i - NC;
+        inputExprs.emplace_back(dim.floorDiv(stride[j]));
+      }
+      outputExprs.emplace_back(dim);
+    }
+
+    SmallVector<AffineMap> indexingMaps = AffineMap::inferFromExprList(
+        {inputExprs, inputExprs, outputExprs}, rewriter.getContext());
+
+    SmallVector<utils::IteratorType> iteratorTypes(
+        outRank, utils::IteratorType::parallel);
+
+    auto computeIndex = [&](OpBuilder &b, Location loc) -> Value {
+      Value ret;
+      for (auto i : llvm::seq<int64_t>(NC, outRank)) {
+        Value idx = b.create<linalg::IndexOp>(loc, i);
+        // Adjust for the pad
+        int64_t pad = padding[i - NC];
+        if (pad != 0) {
+          Value padVal = b.create<arith::ConstantIndexOp>(loc, pad);
+          idx = b.create<arith::SubIOp>(loc, idx, padVal);
```

Comment:
It took me like an hour to figure out why you are doing this. All this is being used for is to see if it matches the value in the indices tensor, so if idx - padVal is negative, it's not actually extracting with that nonsense index or anything, but it clearly won't match the value in the indices tensor, so we know it should be a zero.